### PR TITLE
Dashboard aurora map : Caleb

### LINKED ISSUE #256: Create an aurora overview map
As an astronomer I want an insightful visual so that I can see where auroras are visible in the UK at a given moment

### PR DESCRIPTION
**Pull request by CalebMathers**
This closes #256 

Wrote functions to produce a map of the uk that overlays the likelihood of an aurora sighting, along with the cloud coverage for each county.

### CODE REVIEW

File: dashboard/aurora_map.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
+import geopandas as gpd
+import matplotlib.pyplot as plt
+from psycopg2 import connect, extensions, extras
+from dotenv import load_dotenv
+from os import environ as ENV
+
+COUNTY_MAPPING = {
```
Comment (AlexOBarnes) [inline dashboard/aurora_map.py:7]: Wow caleb doing this by hand is incredible

File: dashboard/aurora_map.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+            key: 'yellow' for key in map_dict if key not in NORTHERN_REGIONS}
+        for region in NORTHERN_REGIONS:
+            colours[region] = 'orange'
+    else:
+        colours = {key: 'orange' for key in map_dict}
+
+    return colours
+
+
+def create_aurora_map() -> plt.figure:
```
Comment (AlexOBarnes) [inline dashboard/aurora_map.py:197]: Good 14 lines

**Review by AlexOBarnes — APPROVED** [review]

File: dashboard/aurora_map.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    "Stoke-on-Trent": "Staffordshire", "Bath and North East Somerset": "Somerset",
+    "Bristol, City of": "Bristol", "North Somerset": "Somerset",
+    "South Gloucestershire": "Gloucestershire", "Plymouth": "Devon", "Torbay": "Devon",
+    "Swindon": "Wiltshire", "Peterborough": "Cambridgeshire",
+    "Luton": "Bedfordshire", "Southend-on-Sea": "Essex",
+    "Thurrock": "Essex", "Medway": "Kent",
+    "Bracknell Forest": "Berkshire", "West Berkshire": "Berkshire",
+    "Reading": "Berkshire", "Slough": "Berkshire",
+    "Windsor and Maidenhead": "Berkshire", "Wokingham": "Berkshire",
+    "Milton Keynes": "Buckinghamshire", "Brighton and Hove": "East Sussex",
```
Comment (olliejth) [inline dashboard/aurora_map.py:27]: Typo

File: dashboard/aurora_map.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    "Vale of Glamorgan": "Vale of Glamorgan", "Cardiff": "Cardiff",
+    "Rhondda Cynon Taf": "Rhondda Cynon Taf", "Caerphilly": "Caerphilly",
+    "Blaenau Gwent": "Blaenau Gwent", "Torfaen": "Torfaen",
+    "Monmouthshire": "Monmouthshire", "Newport": "Newport",
+    "Powys": "Powys", "Merthyr Tydfil": "Merthyr Tydfil"
+}
+
+NORTHERN_REGIONS = ['Scotland', 'North East (England)', 'North West (England)',
+                    'Yorkshire and The Humber', 'Northern Ireland']
+
```
Comment (olliejth) [inline dashboard/aurora_map.py:121]: Why have you split it up into these?

Comment (olliejth) [reply dashboard/aurora_map.py:121]: This has been explained, thanks Caleb.

File: dashboard/aurora_map.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+            JOIN aurora_colour AS ac
+            USING (aurora_colour_id)
+            ORDER BY aa.alert_time DESC
+            LIMIT 1;'''
+
+    with get_connection() as conn:
+        with conn.cursor() as cur:
+            cur.execute(query)
+            result = cur.fetchone()[0]
+
```
Comment (olliejth) [inline dashboard/aurora_map.py:143]: Do you need the "[0]" here?

File: dashboard/shapefile/CTYUA_DEC_2023_UK_BGC.cpg [hunk 6]
```diff
@@ -0,0 +1 @@
+UTF-8
```
Comment (olliejth) [inline dashboard/shapefile/CTYUA_DEC_2023_UK_BGC.cpg:1]: Does this need to be in its own file?

**Review by olliejth — APPROVED** [review]
SOkme minor formatting queries but otherwise good.

**Pull request merged**